herbert_core/classes/instrument_classes/instrument/detectors/@IX_detector_array/IX_detector_array.m

```diff
@@ -74,205 +128,412 @@
+                end
+            end
+            obj.det_bank_ = tmp;
+            
+            if obj.do_check_combo_arg_
+                obj = obj.check_combo_arg();
+            end
         end
-
+        
+        %---------------------------
+        function obj = set.azim (obj, val)
+            ndets = arrayfun (@(O)(O.ndet), obj.det_bank_);
+            % Input must be scalar or an array with ndet elements
+            if ~isnumeric(val) || ~(numel(val)==sum(ndets) || isscalar(val))
+                error('HERBERT:IX_detector_array:invalid_argument',...
+                    ['Azimuthal angle(s) must be a scalar or a ',...
+                    'numeric array with ', num2str(sum(ndets)), ' elements'])
+            end
+            % Alter every bank
+            tmp = obj.det_bank_;
+            if numel(val)>1
+                nend = cumsum(ndets);
+                nbeg = nend - ndets + 1;
+                for i=1:numel(tmp)
+                    tmp(i).azim = val(nbeg(i):nend(i));
+                end
+            else
+                for i=1:numel(tmp)
+                    tmp(i).azim = val;
+                end
+            end
+            obj.det_bank_ = tmp;
+            
+            if obj.do_check_combo_arg_
+                obj = obj.check_combo_arg();
+            end
+        end
+        
+        %---------------------------
+        function obj = set.dmat (obj, val)
+            ndets = arrayfun (@(O)(O.ndet), obj.det_bank_);
+            % Input must have size [3,3,ndet] or [3,3] (==[3,3,1])
+            if ~isnumeric(val) || numel(size(val))>3 || ...
+                    (size(val,1)~=3 || size(val,2)~=3) || all(size(val,3)~=[1,sum(ndets)])
+                if sum(ndets)>1
+                    error('HERBERT:IX_detector_array:invalid_argument',...
+                        ['Detector orientation matrices must be a numeric array ',...
+                        'with size [3,3] or [3,3,', num2str(sum(ndets)), ']'])
+                else
+                    error('HERBERT:IX_detector_array:invalid_argument',...
+                        ['Detector orientation matrix must be a numeric array ',...
+                        'with size [3,3]'])
+                end
+            end
+            % Alter every bank
+            tmp = obj.det_bank_;
+            if size(val,3)>1
+                nend = cumsum(ndets);
+                nbeg = nend - ndets + 1;
+                for i=1:numel(tmp)
+                    tmp(i).dmat = val(:,:,nbeg(i):nend(i));
+                end
+            else
+                for i=1:numel(tmp)
+                    tmp(i).dmat = val;
+                end
+            end
+            obj.det_bank_ = tmp;
+            
+            if obj.do_check_combo_arg_
+                obj = obj.check_combo_arg();
+            end
+        end
+        
+        %---------------------------
+        function obj = set.rotvec (obj, val)
+            ndets = arrayfun (@(O)(O.ndet), obj.det_bank_);
```

Comment:
Done. I added the getter for ndet_bank (returns the variable labelled ndets above) after I'd otherwise completed the class, when I found I kept on needing the number of detectors in each bank in some other algorithms. I'd forgotten that I'd cut-n-pasted the line above in so many of the getters.